### PR TITLE
AUT-3820: Add decryption for jwt on authorize stub endpoint

### PR DESCRIPTION
**Pull request by BeckaL**
This involved:
* Modifying the template.yml to ensure that the relevant environment parameters were picked up from parameters.json when running locally
* Generating a key pair for use locally. These are committed here deliberately, as they are just for local testing use and will not be the same in any other environment
* Using jose to decrypt the request
* Adding a script to more easily generate the request parameter for local testing
* Updating the readme

## How to review
1. Code review
1. Use the instructions in the readme to construct a request parameter. Run the app locally via the readme instructions and see that you hit a sensible looking page (it should just display you a dummy page that shows the unencrypted and decoded payload and data from the algorithm header) and not an error

### CODE REVIEW

File: ipv-stub/src/endpoints/ipv-authorize.ts [hunk 1]
```diff
@@ -34,13 +36,45 @@ async function get(
...
+  vKdO7cFBmDIOASlGgGCEG0PVBC4FJH2pM3FUw72n7YTS+H73Y8ZfTNIgu9K7zxEa
+  mCwimUAKU8Lsjq6Pqa0pZr2rE4l2MfO2j91uCcdlTzdM0kOkwcbzwqEdbDU+FJ4x
+  FT5aaOWyuN/BKFsc5kNz2t4+OaeRu/ev3h7WCqh2MMW5PWDbR2lBnKZR8HvuXZc5
+  ay0dUx098UjkLEBHIyT3FfzhXFMF2ZdOSysDMa64KwqeAWs6tjwM9+Bp3DYLkTsx
+  BML/eqgIwdZI5QBCTxD8YC2oxC1obMspiAoEz05wt8cYscmT4rZAdBGMspuxNqo4
+  6wIDAQAB
+  -----END PUBLIC KEY-----`;
+
+// Step 2: Convert the Base64-encoded data to a Uint8Array for encryption
+  const dataToEncrypt = Uint8Array.from(`eyJhbGciOiJzb21lLWFsZyJ9.eyJmb28iOiJiYXIifQ==.eyJ0aGlzIjoiZG9lc24ndG1hdHRlciJ9`, c => c.charCodeAt(0));
```
Comment (ethanmills) [inline ipv-stub/src/endpoints/ipv-authorize.ts:50]: JWTs shouldn't have padding, you can drop the = at the end https://datatracker.ietf.org/doc/html/rfc7515#section-2:~:text=JWS)%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20May%202015-,Base64url%20Encoding,-Base64%20encoding%20using

File: ipv-stub/src/endpoints/ipv-authorize.ts [hunk 2]
```diff
@@ -34,13 +36,45 @@ async function get(
...
+
+// Step 3: Import the public key
+  const publicKey = await importSPKI(publicKeyPem, "RSA-OAEP-256");
+
+// Encrypt the data
+  const jwe = await new CompactEncrypt(dataToEncrypt)
```
Comment (ethanmills) [inline ipv-stub/src/endpoints/ipv-authorize.ts:56]: What's this for?

Comment (BeckaL) [reply ipv-stub/src/endpoints/ipv-authorize.ts]: sorry left in by accident just taken out

File: ipv-stub/src/endpoints/ipv-authorize.ts (hunk 1 above)
Comment (BeckaL) [reply ipv-stub/src/endpoints/ipv-authorize.ts]: This also left in by accident, out now

**Review by alhcomer — APPROVED** [review]

**Pull request merged**